src/kOS.Safe/Compilation/Opcode.cs

```diff
@@ -1141,14 +1165,17 @@ public class OpcodeLogicToBool : Opcode
 
         public override void Execute(ICpu cpu)
         {
+            // This may look like it's just pointlessly converting from a
+            // ScalarBoolean to a primitive boolean and then back into a
+            // ScalarBoolean, and in the case where the operand was already
+            // a ScalarBoolean that would be true.  But the purpose of this opcode
+            // is to also change integers and floats into booleans. Thus the call to
+            // Convert.ToBoolean():
```

Comment:
Based on our conversation in slack the other day, I didn't think that using numbers as boolean values was supported.  But working with this branch, the following test works:
```
set x to 1.
if x and true { print "hello world". } // print's "hello world"
```
We could probably streamline this code, and make it less confusing, by using `Convert.ToType` and enabling a conversion to `BooleanValue` from `ScalarValue`.

But you and I had talked about omitting the feature, so I didn't want to update it as part of the review without discussing it first.